context/response/responses.go

```diff
@@ -0,0 +1,53 @@
+package response
+
+import (
+	beecontext "github.com/astaxie/beego/context"
+)
+
+// JSON renders value to the response as JSON
+func JSON(value interface{}, encoding ...bool) Renderer {
```

Comment:
OK, I refactored the code a bit and hopefully it will look better now.
One thing I want to keep in mind: I don't want controller code to rely on c.Ctx and want it to be as "stateless" as possible. The reason is that it makes testing the controller method much easier. 
No need to setup context to test the method if it simply accepts all parameters as function parameters and returns simple objects
Please review now and let me know if this is better